rsvim_core/src/state/fsm/normal.rs

```diff
@@ -145,17 +165,40 @@ impl NormalStateful {
   }
 }
 
+impl NormalStateful {
+  pub fn new_line(&self, data_access: &StatefulDataAccess) {
+    let tree = data_access.tree.clone();
+    let mut tree = lock!(tree);
+    let current_window = tree.current_window_mut().unwrap();
+    let current_window_id = current_window.id();
+    let buffer = current_window.buffer().upgrade().unwrap();
+    let mut buffer = lock!(buffer);
+    let eol = CompactString::new(format!("{}", buffer.options().end_of_line()));
+
+    cursor_ops::cursor_insert(&mut tree, current_window_id, buffer.text_mut(), eol);
+  }
+}
+
 impl NormalStateful {
   /// Cursor move in current window, with buffer scroll.
   pub fn cursor_move(&self, data_access: &StatefulDataAccess, op: Operation) -> StatefulValue {
+    self.cursor_move_include_eol(data_access, op, false)
+  }
+
+  pub fn cursor_move_include_eol(
```

Comment:
Let's change this internal function signature to:

```rust
fn _cursor_move_impl(
```

1. Remove the "pub" keyword, as it is internal.
2. Rename the "include_eol" to "impl", this is quite personal taste.... :)